services/organisation/organisation.go

```diff
@@ -72,3 +72,19 @@ func CreateOrganisation(req models.CreateOrgRequestModel, db *gorm.DB, userId st
 
 	return &org, nil
 }
+
+func DeleteOrganisation(orgId string, userId string, db *gorm.DB) error {
+	var org models.Organisation
+	if err := db.Where("id = ? AND deleted = ?", orgId, false).First(&org).Error; err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return errors.New("organisation not found")
+		}
+		return err
+	}
```

Comment:
this should be in the model, also you should have repository function for those db calls

Reply:
Alright, noted. On it

Reply:
I've effected the change. Thank you